pkg/kubelet/kubelet.go

```diff
@@ -1701,6 +1709,50 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	return nil
 }
 
+func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod) {
+	podResized := false
+	for _, container := range pod.Spec.Containers {
+		if len(diff.ObjectDiff(container.ResourcesAllocated, container.Resources.Requests)) > 0 {
```

Comment:
just of my curioucity,  should this diff only check the resources that can be resized or resize is supported, ie. CPU, Memory for the current design ? so it won't be disturbed by other resources such GPU or other cusotmer resources ?

Reply:
it checks if any of the desired resource values have changed than what's allocated. currently there's cpu, memory, and ephemeral-storage supported as valid resource-types in Resources field. someone (from alibaba i think) asked about implementing this for ephemeral storage as well but i scoped it out to keep things simple.

Reply:
ephemeral resources are fine with resizing - the others like GPU etc, could be a valid resource for resizing. probably can have a future task for those resources.

Reply:
they would require pods to be restarted by the controller - same behavior as today. the scope of this KEP is to allow restart-free resizing only for CPU and memory. ephermeral storage and other extended resources are explicitly disallowed in this kep and i have implemented validation check to block it. So you will never be able to trigger this by changing anything besides cpu or memory.